components/engine-camel/src/main/java/org/eclipse/dirigible/components/engine/camel/processor/CamelProcessor.java

```diff
@@ -76,4 +81,22 @@ private void removeAllRoutes() {
             throw new RuntimeException(e);
         }
     }
+
+    public Object invokeRoute(String routeId, Object payload, Map<String, Object> headers) {
+        try (FluentProducerTemplate producer = context.createFluentProducerTemplate();) {
+            return producer
+                    .withHeaders(headers)
+                    .withBody(payload)
+                    .to(routeId)
+                    .request();
+        } catch (IOException e) {
+            throw new RuntimeException("Could not create fluent producer");
+        }
+
+        // For future reference. Alternative way to invoke route:
```

Comment:
This should be documented somewhere else

Reply:
It is removed from the code and i just added it at the bottom of this PR's comment just so we have it in the feature history